### PR TITLE
fixes and cleanup for pre-0.7 work - "butter risky"

### PR DESCRIPTION
**Pull request by jdknight**
The following contains a series of improvements for the Confluence builder extension. Most notables includes:

- Corrects slew of markup/publishing errors observed in #39, #40 and #41 (see 255043422d25f8e437baf3182eafa6464a1ab288 and b6bb29230db22ccc45a5852c75f88ef4cfbea422).
- Corrects unexpected "garbage" encoding data in #52 (see fa206b18982ad7765fd23e57b47718c6795df5ac).
- Have title references emphasized (instead of strong'ed).

Perform validation set testing on Confluence 6.0.6 server (REST and XML-RPC) and Confluence Cloud  (REST and XML-RPC) with expected/improved results.

### CODE REVIEW

File: sphinxcontrib/confluencebuilder/common.py [hunk 1]
```diff
@@ -75,3 +76,13 @@ def verbose(*args, **kwargs):
     def warn(*args, **kwargs):
         if ConfluenceLogger.logger:
             ConfluenceLogger.logger.warning(*args, **kwargs)
+
+    @staticmethod
+    def trace(title, data):
+        try:
+            with io.open('trace.log', 'a', encoding='utf-8') as file:
+                file.write(u'[%s]\n' % title)
```
Comment (tonybaloney) [inline sphinxcontrib/confluencebuilder/common.py]: this could cause issues in py2.7

```
u'%s' % unicode('👉')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xf0 in position 0: ordinal not in range(128)
```

Comment (tonybaloney) [reply sphinxcontrib/confluencebuilder/common.py]: that is, assuming title is a byte string. I would encode it first

File: sphinxcontrib/confluencebuilder/translator/wiki.py [hunk 2]
```diff
@@ -26,10 +26,10 @@
 
 # Confluence encoding for special chars
 SPECIAL_VALUE_REPLACEMENTS = {
-    ('{', '&lcub;'),
-    ('}', '&rcub;'),
-    ('<', '&lt;'),
-    ('>', '&gt;')
+    ('{', '&#123;'),
```
Comment (tonybaloney) [inline sphinxcontrib/confluencebuilder/translator/wiki.py]: why was this changed? I remember having issues with code points.

File: sphinxcontrib/confluencebuilder/common.py (hunk 1 above)
Comment (jdknight) [reply sphinxcontrib/confluencebuilder/common.py]: Ah, I don't think I fully tested tracing in 2.7. I'll take a look and cleanup.

File: sphinxcontrib/confluencebuilder/translator/wiki.py (hunk 2 above)
Comment (jdknight) [reply sphinxcontrib/confluencebuilder/translator/wiki.py]: Interesting. Before the change (from commit b6bb29230db22ccc45a5852c75f88ef4cfbea422), I got an error when using the curly brackets:
```
MSG: Error parsing xhtml: Undeclared general entity "lcub"
 at [row,col {unknown-source}]: [1,390]
```
I'll go back and try to re-produce the issue. I also wonder if any of this relates to the use of the `HTMLParser` (proposed for removal in 255043422d25f8e437baf3182eafa6464a1ab288).

File: sphinxcontrib/confluencebuilder/common.py (hunk 1 above)
Comment (jdknight) [reply sphinxcontrib/confluencebuilder/common.py]: For this trace method, the title value is an internal-only provided (i.e. not a document title). I wouldn't mind just leaving the method as-is and just assume a provided title is a compatible byte string for the running interpreter.

File: sphinxcontrib/confluencebuilder/translator/wiki.py (hunk 2 above)
Comment (jdknight) [reply sphinxcontrib/confluencebuilder/translator/wiki.py]: Do you have any hints on your observed issues using decimal values? After trying to toggle the change, I always see an error generated when using the '\&lcub;' (and respective) values:

```
An unsupported Confluence API call has been made.

REQ: PUT
RSP: 400
URL: https://jdknight.atlassian.net/wiki/rest/api/
API: content
MSG: Error parsing xhtml: Undeclared general entity "lcub"
 at [row,col {unknown-source}]: [3,30]
```

Always observing errors in 2.7, 3.4-3.6 on both Confluence Server and Confluence Cloud when using '\&lcub;'. No issues with the decimal variant. Thoughts?

**Comment by tonybaloney** [comment]
@jdknight ready for release

**Comment by jdknight** [comment]
@tonybaloney Thanks! I'll do some re-testing to re-sync myself, take a look at the recent issues to see if there's anything critical and if everything good I can ping you to wrap up the owner tasks in #29 (if you thin that's a good plan).
